src/parser/Parser.ts

```diff
@@ -2911,6 +2936,19 @@ export class Parser {
         return this.tokens[this.current - 1];
     }
 
+    /**
+     * There are sometimes we get catch an error that is a diagnostic
+     * If that's the case, we want to continue parsing
+     * Otherwise, re-throw the error
```

Comment:
```suggestion
     * Sometimes we catch an error that is a diagnostic.
     * If that's the case, we want to continue parsing.
     * Otherwise, re-throw the error
```